File: chrome/common/extensions/docs/server2/environment.py
# Copyright 2013 The Chromium Authors. All rights reserved.
# Use of this source code is governed by a BSD-style license that can be
# found in the LICENSE file.

import os
import sys


def _IsServerSoftware(name):
  return os.environ.get('SERVER_SOFTWARE', '').find(name) == 0


def IsDevServer():
  return _IsServerSoftware('Development')


def IsReleaseServer():
  return _IsServerSoftware('Google App Engine')


def IsPreviewServer():
  return sys.argv and os.path.basename(sys.argv[0]) == 'preview.py'

File: mojo/public/environment/standalone/environment.cc
// Copyright 2014 The Chromium Authors. All rights reserved.
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

#include "mojo/public/environment/environment.h"

#include "mojo/public/environment/standalone/buffer_tls_setup.h"
#include "mojo/public/utility/run_loop.h"

namespace mojo {

Environment::Environment() {
  internal::SetUpCurrentBuffer();
  RunLoop::SetUp();
}

Environment::~Environment() {
  RunLoop::TearDown();
  internal::TearDownCurrentBuffer();
}

}  // namespace mojo
